insonmnia/hub/state.go

```diff
@@ -0,0 +1,109 @@
+package hub
+
+import (
+	log "github.com/noxiouz/zapctx/ctxlog"
+	"github.com/sonm-io/core/insonmnia/resource"
+	"go.uber.org/zap"
+)
+
+type hubState struct {
+	Tasks    map[string]*TaskInfo
+	Devices  map[string]DeviceProperties
+	AskPlans AskPlansData
+	Acl      []string
+	Deals    map[DealID]*DealMeta
+	Orders   map[OrderID]ReservedOrder
+	Miners   map[string]map[OrderID]resource.Resources
+}
+
+func (h *Hub) dumpState() error {
+	state := h.getState()
+
+	if err := h.cluster.Synchronize(state); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (h *Hub) loadState(state *hubState) error {
```

Comment:
this is a method of Hub, why is it defined in other file? Again this is one vote for making a state a member of hub and defining methods of that state